### PR TITLE
Add robot init function to create *ri* and robot instance

### PR DESCRIPTION
**Pull request by snozawa**
Add robot init function to create `*ri*` and robot instance according to the following discussion:
https://github.com/jsk-ros-pkg/jsk_pr2eus/issues/175

* Behavior
     1. Without /robot/type and argument, or no interface file is found, robot instance and interface instance are not created and errors are invoked.   
     `(robot-init) ;; without ros param` -> error.
     2. If argument are specified and robot interface file is found, robot instance and interface with given name are created.   
     `(robot-init "pr2")` -> OK
     3. If no argument is specified, /robot/type ROSPARAM is set, and robot interface file is found, robot instance and interface with given name are created.  
     `(robot-init) ;; with /robot/type setting` -> OK. Create `*ri*` and robot from /robot/type.

* Typical usage:
```
     (robot-init) ;; With setting /robot/type ROSPARAM anywhere. 
     (robot-init (or (ros::get-param \"/robot/type\") \"pr2\")) ;; If /robot/type ROSPARAM is specified, use it. Otherwise, use \"pr2\" by default.
```

* Configuring user-defined robot:      
     This function searches robot interface file from rospack plugins. 
     If user want to use their own robots from robot-init function,   
     please write export tag in [user_defined_rospackage]/package.xml as pr2eus/package.xml.     
```
  <export>
    <pr2eus robot-name="pr2" interface-file="${prefix}/pr2-interface.l" />
 </export>
```

I also add test code for it (robot-init-test.l).

### CODE REVIEW

**Comment by snozawa** [comment]
Travis passed thanks for https://github.com/jsk-ros-pkg/jsk_pr2eus/pull/223.

**Comment by k-okada** [comment]
I think you'd better to leave this PR open for a while, if you want promote this new way of initialize `*ri*` and `*robot*`, otherwise no one realize this changes...

- where ` piped-fork-returns-list ` defined ?
- is calling `rospack` from piped fork safe? (c.f. http://www.jsk.t.u-tokyo.ac.jp/redmine/public/issues/134)
- do we have to write export tag for all robots? ened todo list?

**Comment by snozawa** [comment]
> I think you'd better to leave this PR open for a while, if you want promote this new way of initialize *ri* and *robot*, otherwise no one realize this changes...

This PR is discussed more than a half year, 
https://github.com/jsk-ros-pkg/jsk_pr2eus/issues/175
and I waited as your comments in almost one day.
But I'm very sorry for too short period to leave this PR open.

> where piped-fork-returns-list defined ?

Defined in jskeus : https://github.com/euslisp/jskeus/blob/master/irteus/irtutil.l#L369.
This is the commit at sourforge age.
This performs
- Close p-opened stream after using
- Returns list

```
11.irteusgl$ ls -l /tmp/irtmodel-ik-14751/
合計 12
-rw-r--r-- 1 leus leus 1347 Apr 27 09:33 hrp2jsknts-robot-2016-04-27-09-33-51-95-16-97-failure.l
-rw-r--r-- 1 leus leus 1351 Apr 27 09:33 hrp2jsknts-robot-2016-04-27-09-33-51-99-51-47-failure.l
-rw-r--r-- 1 leus leus 1277 Apr 27 09:33 hrp2jsknts-robot-2016-04-27-09-33-52-03-33-48-failure.l
0
```
```
13.irteusgl$ pprint (piped-fork-returns-list "ls -l /tmp/irtmodel-ik-14751/")
("合計 12"
 "-rw-r--r-- 1 leus leus 1347 Apr 27 09:33 hrp2jsknts-robot-2016-04-27-09-33-51-95-16-97-failure.l"
 "-rw-r--r-- 1 leus leus 1351 Apr 27 09:33 hrp2jsknts-robot-2016-04-27-09-33-51-99-51-47-failure.l"
 "-rw-r--r-- 1 leus leus 1277 Apr 27 09:33 hrp2jsknts-robot-2016-04-27-09-33-52-03-33-48-failure.l")
nil
```

> is calling rospack from piped fork safe? (c.f. http://www.jsk.t.u-tokyo.ac.jp/redmine/public/issues/134)

Oh, is this resolved by ueda's commit??


For this PR, originally I did not use `piped-fork` and I used `ros::rospack-plugins`.
However, `ros::rospack-plugins` euslisp function is too new
https://github.com/jsk-ros-pkg/jsk_roseus/commit/e538fdd9d1008ca27c137578086380c9747356db.
So we could not use `ros::rospack-plugins` from `ros-hydro-roseus` because of ros hydro deb freeze.
Moreover, `jsk_pr2eus`'s hydro travis test failed because of this reason.
Therefore, I reluctantly add `piped-fork` for robot-init.

> do we have to write export tag for all robots? ened todo list?

Yes, if we want to initialize our own robot from `robot-init`, 
- Just call `(robot-init "pepper")`
- Call `(robot-init)` with writing export tag for robot-defined rso package.

Currently, hrpsys_ros_bridge_tutorials and pr2eus have this export tag.

**Comment by k-okada** [comment]
> 
> This PR is discussed more than a half year,

This PR is leave more than a half year, without any discussion. This means
- no one feel this is required
- no one noticed 
- every one agreed but too shy to say +1

**Comment by snozawa** [comment]
> - no one feel this is required 
> - no one noticed
> - every one agreed but too shy to say +1

Exactly, as your comment, all of these are correct.

The purpose of this PR is 
- Enable to select robot model and ri without writing hard-coded `if` or `cond` : https://github.com/jsk-ros-pkg/jsk_pr2eus/issues/175#issue-106967061.

In addition to your comments, 
- Some people  think this functionality is required, but write similer code everywhere.
    - Almost same codes  
      https://github.com/jsk-ros-pkg/jsk_control/blob/master/jsk_footstep_controller/euslisp/util.l  
      https://github.com/jsk-ros-pkg/jsk_control/blob/master/contact_states_observer/euslisp/contact-states-observer.l  
      https://github.com/jsk-ros-pkg/jsk_demos/blob/master/jsk_2015_06_hrp_drc/drc_task_common/euslisp/robot-util.l  
       https://github.com/jsk-ros-pkg/jsk_demos/blob/master/jsk_2015_06_hrp_drc/drc_task_common/euslisp/door-open/create-robot-instance.l  
       https://github.com/jsk-ros-pkg/jsk-ros-pkg-unreleased/blob/master/hrp2_controllers/hrp2_cmd_vel/src/object-reaching-releasing-motion-handler.l  
    - Similar codes  
      https://github.com/jsk-ros-pkg/jsk_control/blob/master/jsk_calibration/euslisp/hrp2-calibration.l  
      https://github.com/jsk-ros-pkg/jsk_control/blob/master/jsk_ik_server/euslisp/ik-server-impl/jaxon_red-ik-server.l  
      https://github.com/jsk-ros-pkg/jsk_control/blob/master/jsk_ik_server/test/fullbody-ik-client-test.l  
      https://github.com/jsk-ros-pkg/jsk-ros-pkg-unreleased/blob/master/jsk_robots/khr_driver/euslisp/jaxon_red-khr-masterslave.l  

- Some people unexpectedly load the above-mentioned similer code (for example, silimer code in drc_task_common are used for drc programers.)
- If students use just one robot, they do not think this functionality is required because they do not change robot for their programs.

File: pr2eus/robot-interface.l [hunk 1]
```diff
@@ -1683,4 +1683,45 @@ send-action [ send message to action server, it means robot will move ]"
...
+     This function searches robot interface file from rospack plugins.
+     If user want to use their own robots from robot-init function,
+     please write export tag in [user_defined_rospackage]/package.xml as pr2eus/package.xml.
+   "
+  (let* (;; Check existence ros::rospack-plugins for robot-name-list and interface-file-list.
+         ;;    For example, old deb environment before ros::rospack-plugins is committed.
+         (robot-name-list
+          (if (fboundp 'ros::rospack-plugins)
+              (mapcar #'cdr (ros::rospack-plugins "pr2eus" "robot-name"))
+            (piped-fork-returns-list "rospack plugins --attrib=robot-name pr2eus | cut -d\\  -f2")))
```
Comment (k-okada) [inline pr2eus/robot-interface.l]: roseus < 1.4.0 does not support `ros::rospack-plugins`

Comment (snozawa) [inline pr2eus/robot-interface.l]: Yes, so i used fboundp to check existence.

**Pull request merged**